agent/agent.go

```diff
@@ -373,15 +373,21 @@ func New(bd BaseDeps) (*Agent, error) {
 		cache:           bd.Cache,
 	}
 
-	cacheName := cachetype.HealthServicesName
-	if bd.RuntimeConfig.UseStreamingBackend {
-		cacheName = cachetype.StreamingHealthServicesName
+	// TODO: create rpcClientHealth in BaseDeps once NetRPC is available without Agent
+	conn, err := bd.GRPCConnPool.ClientConn(bd.RuntimeConfig.Datacenter)
+	if err != nil {
+		return nil, err
```

Comment:
What does it mean for this to error here? Should we add extra context? Would this fail if there was a connectivity issue with other consul agents. If so that seems bad to fail to startup.

Reply:
Good question. As long as we don't use `WithBlock()` (we don't use and it seems unlikely that we ever would) then the only errors that `grpc.Dial` can return are errors related to the options we pass into `grpc.Dial`, or building the resolver. Our customer resolver builder does not return errors, so this seems ok.  I can definitely add some context to the error.

I'll see if there is another place that this could be done.

Reply:
It looks like the alternative would be to pass the `GRPCConnPool` to `health.MaterializerDeps` instead of the `conn`. That would however force a separate connection for each Materializer in the Store, and makes closing the connections more difficult because the lifecycle is tied to the materialize. The current approach seems correct based on what I've found so far.